runtime/gc_base/RootScanner.cpp

```diff
@@ -536,7 +537,23 @@ MM_RootScanner::scanOneThread(MM_EnvironmentBase *env, J9VMThread* walkThread, v
 		doVMThreadSlot(slot, &vmThreadIterator);
 	}
 
-	GC_VMThreadStackSlotIterator::scanSlots((J9VMThread *)env->getOmrVMThread()->_language_vmthread, walkThread, localData, stackSlotIterator, isStackFrameClassWalkNeeded(), _trackVisibleStackFrameDepth);
+	J9VMThread *currentThread = (J9VMThread *)env->getOmrVMThread()->_language_vmthread;
+	GC_VMThreadStackSlotIterator::scanSlots(currentThread, walkThread, localData, stackSlotIterator, isStackFrameClassWalkNeeded(), _trackVisibleStackFrameDepth);
+
+#if JAVA_SPEC_VERSION >= 19
+	if (NULL != walkThread->currentContinuation)
+	{
+		bool const compressed = _extensions->compressObjectReferences();
+		j9object_t carrierThreadObject = walkThread->carrierThreadObject;
+		MM_ForwardedHeader forwardHeader(carrierThreadObject, compressed);
+		if (forwardHeader.isForwardedPointer()) {
```

Comment:
Typically, carrierThreadObject should have been fixed by now by preceding step that fixes up thread struct slots. As we discussed, the only reason we think this is possible is because of recently remembered optimization, that may defer the fixup. It will occur if  carrierThreadObject just got tenured. Let's confirmed, that's the case.

Either way, this is not an acceptable fix (we cannot have scavenger specific code in a middle of common root scanner). We could do override  this method (or part of it) in ScavengerRootScanner and do that extra check only there.